_docs/cardano/2017-01-01-proof-of-stake.md

```diff
@@ -9,83 +9,173 @@ visible: true
+during the same epoch.
+
+### Multiparty Computation
+
+The fundamental problem of election process is its unbiasedness. We need some randomness
+as a base for election, in this case results of this election will be random and fair.
+So the question is where this randomness can be obtained from?
+
+We are using multiparty computation (MPC) approach: each elector independently performs an
+action which can be called "coin tossing" and after that shares result with other electors.
+The idea is that results are randomly generated by each elector, but eventually they agree on
+the same final value.
+
+#### Commitment Phase
+
+First of all, elector generates a secret (special random value). Then elector forms a
+"commitment" which is a message that contains encrypted shares and proof of secret.
+
+Then elector signs this commitment with its secret key, specifies epoch's number and attaches
+its public key. In this case everybody can check who created this commitment and which epoch
+this commitment is for.
+
+After that elector send its commitment to other electors, so eventually each elector collects
+commitments from all other electors.
+
+#### Opening Phase
+
+Now elector sends an "opening", special value for opening a commitment. You can think of a
+commitment as a locked box (with a secret in it), and opening is a key we need to open this
+box and get the secret from it.
+
+So eventually elector has commitments and openings. Then elector verifies that commitments
+and openings are match, and if so, he gets the secrets from commitment and forms a seed from
+these secrets. So all electors get the same seed, and it will be used for Follow the Satoshi
+algorithm.
+
+### Follow the Satoshi
+
+After electors get the seed (randomness we need), they have to select particular slot leaders for
+the next epoch. This is where Follow the Satoshi (FTS) algorithm came in. It can be shown like
+this:
+
+```
+         +-----+
+SEED --->| FTS |---> ELECTED_SLOT_LEADERS
+         +-----+
+```
+
+Let's elaborate a little bit on how a slot leader gets selected. The smallest, atomic piece
+of value is a coin, we call it "[Lovelace](/glossary/#lovelace)". Fundamentally, we can say
+that the ledger produces distribution of coins, and since slot leaders can be selected from
+stakeholders only, we are talking about distribution of stake. FTS is an algorithm that
+verifiably picks a coin, and when coin owned by stakeholder `S` gets selected, `S` become a
+slot leader. It is obviously that the more coins `S` has, the higher probability that one of
+his coins will be picked.
+
+The reason why does it called "Follow the Satoshi" is that in Bitcoin, atomic piece of currency
+is called "Satoshi", honoring Satoshi Nakamoto, the creator of Bitcoin. 
+
+## Honest Majority
+
+The fundamental assumption of a protocol is a **honest majority**. It means that
+the most of protocol participants (i.e. at least 50% + 1) are honest ones. In
```

Comment:
Not `50% + 1` participants (participants are nodes), but participants owning `50% + 1` of stake. It can be only two participants each holding 30% of coins.